src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -90,12 +90,26 @@ public async virtual Task<HostSecretsInfo> GetHostSecretsAsync()
                         _logger.LogDebug("Loading host secrets");
 
                         hostSecrets = await LoadSecretsAsync<HostSecrets>();
-                        if (hostSecrets == null)
+                        try
+                        {
+                            if (hostSecrets == null)
```

Comment:
We also have the same load/generate/persist flow for Function secrets as well [here](https://github.com/Azure/azure-functions-host/blob/1715c505237c0ccc8de8b16fd80b9d16bcda37c3/src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs#L168). Do the error logs show that we're also getting races there as well? If so we probably want to make similar changes in both codepaths.

Reply:
Made the changes!